src/docs/arch-s3-extension.dox

```diff
@@ -0,0 +1,79 @@
+/*! @arch_page arch-s3-extension S3 Storage Source
+
+@section s3_extension_storage_source S3 Storage Source Implementation
+
+S3 storage source, or the S3 store, is a WiredTiger extension that implements the storage source
+abstraction for AWS's S3 object-store. The store extends the WiredTiger's capability to store the
+data files on the S3 cloud. The extension provides a means to construct, configure and control a
+custom file system linked to an S3 bucket. The extension internally uses the API provided by the
+AWS C++ SDK to interact with the S3 cloud.
+
+The S3 store implements the following key functionalities defined by the storage source interface:
+
+Initialize the store, the AWS SDK and create a logging module for the store and the SDK.
```

Comment:
Should we have a bullet point list from this paragraph to list the different functionalities?

Reply:
Done